### PR TITLE
Fix for issue #586

### PR DESCRIPTION
**Pull request by silabs-oysteink**
Updated description of mtvec reset and initialization.
Other CSR reset values seems clear to me, those with dependencies on various parameters or input values are described.

### CODE REVIEW

File: docs/user_manual/source/control_status_registers.rst [hunk 1]
```diff
@@ -668,7 +668,7 @@ Detailed:
   | 1:0     | WARL (0x0, 0x1)  | **MODE**: Interrupt handling mode. 0x0 = non-vectored CLINT mode, 0x1 = vectored CLINT mode.                  |
   +---------+------------------+---------------------------------------------------------------------------------------------------------------+
 
-The initial value of ``mtvec`` is equal to {**mtvec_addr_i[31:7]**, 5'b0, 2'b01}.
+Out of reset ``mtvec`` has the value of 32'h00000001. This value is not observable by SW as ``mtvec`` is initialized to {**mtvec_addr_i[31:7]**, 5'b0, 2'b01} when ``fetch_enable_i`` is asserted the first time after reset.
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst]: after reset -> after reset release

File: docs/user_manual/source/control_status_registers.rst [hunk 2]
```diff
@@ -710,7 +710,7 @@ Detailed:
   | 1:0     | WARL (0x3)       | **MODE**: Interrupt handling mode. Always CLIC mode.                                                          |
   +---------+------------------+---------------------------------------------------------------------------------------------------------------+
 
-The initial value of ``mtvec`` is equal to {**mtvec_addr_i[31:7]**, 1'b0, 6'b000011}.
+Out of reset ``mtvec`` has the value of 32'h00000003. This value is not observable by SW as ``mtvec`` is initialized to {**mtvec_addr_i[31:7]**, 1'b0, 6'b00000011} when ``fetch_enable_i`` is asserted the first time after reset.
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:713]: after reset -> after reset release
6'b00000011 -> 6'b000011

**Review by Silabs-ArjanB — CHANGES_REQUESTED** [review]

File: docs/user_manual/source/control_status_registers.rst (hunk 1 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/control_status_registers.rst]: Fixed already (?)

File: docs/user_manual/source/control_status_registers.rst (hunk 2 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/control_status_registers.rst]: Fixed

**Pull request merged**